File: src/985. Sum of Even Numbers After Queries.cpp
class Solution {
public:
    vector<int> sumEvenAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
        int S = 0;
        vector<int> ans;
        for (auto& num : nums) {
            if (num % 2 == 0)
                S += num;
        }
        for (auto& query : queries) {
            int val = query[0], i = query[1];
            if (nums[i] % 2 == 0)
                S -= nums[i];
            nums[i] += val;
            if (nums[i] % 2 == 0)
                S += nums[i];
            ans.push_back(S);
        }
        return ans;
    }
};

File: src/985. Sum of Even Numbers After Queries.py
class Solution:
    def sumEvenAfterQueries(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        S = sum(num for num in nums if num % 2 == 0)
        ans = []
        for val, i in queries:
            if nums[i] % 2 == 0:
                S -= nums[i]
            nums[i] += val
            if nums[i] % 2 == 0:
                S += nums[i]
            ans.append(S)
        return ans